lib/finsta_web/live/post_live/index.ex

```diff
@@ -74,16 +68,9 @@ defmodule FinstaWeb.PostLive.Index do
   @impl true
   def handle_event("delete", %{"id" => id}, socket) do
     post = Posts.get_post!(id)
+    {:ok, _} = Posts.delete_post(socket.assigns.current_user, post)
```

Comment:
What about passing the post id instead of the post data to `Posts.delete_post`? In this way, you save a query and avoid loading data which does not belongs to user at all.

Reply:
I agree with this but I've decided not to change it since it's the way it was built by the Phoenix generator.

However, now I'm convinced it's better to do it the way you suggested (and it wouldn't be hard to explain why we are introducing this change)